packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -45,7 +48,7 @@ export class DataProxyEngine extends Engine {
     this.logEmitter.on('error', () => {})
 
     const [host, apiKey] = this.extractHostAndApiKey()
-    this.remoteClientVersion = getClientVersion(this.config)
+    this.remoteClientVersion = P.then(() => getClientVersion(this.config))
```

Comment:
Always resolves immediately, unless it's an integration version client, or a local one.

Reply:
We defer the promise resolution by wrapping it in another promise.